File: frontend/src/api/copilot.js
/**
 * Copilot API client.
 *
 * Calls POST /api/copilot/ask to interact with the AI assistant.
 */

const API_BASE = '/api';

export async function askCopilot(message, conversationHistory = [], storeUrl = null, category = 'Home & Kitchen') {
  const url = `${API_BASE}/copilot/ask`;
  const res = await fetch(url, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
    },
    body: JSON.stringify({
      message,
      conversation_history: conversationHistory,
      store_url: storeUrl,
      category: category,
    }),
  });

  if (!res.ok) {
    const errText = await res.text().catch(() => '');
    throw new Error(`Copilot request failed (${res.status}): ${errText}`);
  }

  return res.json();
}

File: backend/app/models/copilot.py
"""
Pydantic models for the AI Copilot Assistant feature.
"""

from __future__ import annotations

from typing import Any, Literal
from pydantic import BaseModel, Field


class CopilotMessage(BaseModel):
    """Single chat message in conversation history."""
    role: Literal["user", "assistant"]
    content: str


class CopilotAskRequest(BaseModel):
    """Payload for POST /api/copilot/ask."""
    message: str = Field(..., description="User's question or command")
    conversation_history: list[CopilotMessage] = Field(
        default_factory=list,
        description="Previous conversation messages (capped at ~10 by client/server)",
    )
    store_url: str | None = Field(default=None, description="Active merchant store domain")
    category: str | None = Field(default="Home & Kitchen", description="Current selected category")


class CopilotAskResponse(BaseModel):
    """Response returned by POST /api/copilot/ask."""
    reply: str = Field(..., description="Grounded AI response text")
    suggested_proposal: dict[str, Any] | None = Field(
        default=None,
        description="Optional structured proposal payload if the AI generated a concrete recommendation",
    )
